File: Client/src/app/models/ArtPiece.ts
import { ArtPieceType } from "./ArtPieceType";
import { Medium } from "./Medium";

export interface ArtPiece {
  id: number;
  title: string;
  description?: string;
  createdAt: Date;
  mediums: Medium[];
  type: ArtPieceType;
}

File: art-portfolio-api/Models/DomainModels/ArtPiece.cs
namespace art_portfolio_api.Models.DomainModels
{
    public class ArtPiece
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public DateOnly CreatedAt { get; set; }
        public List<Medium> Mediums { get; set; }
        public ArtPieceType Type { get; set; }
    }
}